### ISSUE TITLE
Terraform does not bubble up VPC Peering errors

### ISSUE THREAD

**Issue by kvz** [hashicorp/terraform#8933]
Hi there,

Running terraform 0.7.3, and when VPC CIDRs overlap when trying to set up a peering connection, it terraform is not bubbling up the error:

```bash
Error applying plan:

1 error(s) occurred:

* aws_vpc_peering_connection.us-east-1-eu-west-1: Error waiting for VPC Peering Connection (pcx-cfb33ba6) to become available: unexpected state 'failed', wanted target 'pending-acceptance'. last error: %!s(<nil>)

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

Amazon does show me why it failed to some extend: 

![screen shot 2016-09-20 at 14 59 37](https://cloud.githubusercontent.com/assets/26752/18671064/e2cc6924-7f42-11e6-86cb-845a086f65e3.png)

Certainly not a blocking issue for me, but thought I'd share.

**Comment by antonbabenko** [comment]
Correct me if I am wrong, but the resource id is `aws_vpc_peering_connection.us-east-1-eu-west-1`. Are you trying to peer VPCs in different regions ? If so, it is not allowed by AWS.

**Comment by kvz** [comment]
Yes it was totally my fault. I thought thought the error could be bubbled up as more folks my be doing silly things like me :)

**Comment by kwilczynski** [comment]
@kvz hi there! This is a good idea! Will look into it.

**Comment by kwilczynski** [comment]
This is is proving to be more complex (implementation wise) that I initially anticipated. Why? Because this particular "issue" is not really an error returned by the API layer, but rather the state of the VPC peering connection (which is "failed" in this case). If we return this as an error, then due to how code approaches error handling, we will then interrupt execution and any attempt on refreshing the state of the connection as long as it is in this bad state (before AWS recycles it eventually; you cannot delete failed VPC peering connection, sadly).

I need to think a little bit more about this.

**Comment by stack72** [comment]
Closed via #9038
